builder/typescript/main.ts

```diff
@@ -0,0 +1,13 @@
+import {NewStudentLessonBuilder} from "./new-student-lesson-builder";
+import {LessonDirector} from "./lesson-director";
+
+function main() {
+  const lessonBuilder = new NewStudentLessonBuilder();
+
+  const lessonDirector = new LessonDirector(lessonBuilder);
+  lessonDirector.make();
+
+  const lesson = lessonBuilder.getResult();
+
+  console.log(`${lesson.name} - ${lesson.price} - ${lesson.discountedPrice}`);
```

Comment:
Belki bir alt satıra diğer örneklerde olduğu gibi output satırı da bırakılabilir, önemli değil ama olsa fena olmaz gibi.
// output: Artificial Intelligence -  Beginner to Advanced in 10 Minute. - 49,99 - 24,995

Reply:
TypeScript, Python ve C# örneklerine eklendi 👍